flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/runtime/stream/sql/StreamFileSystemITCaseBase.scala

```diff
@@ -23,17 +23,20 @@ import org.apache.flink.table.api.TableEnvironment
 import org.apache.flink.table.api.bridge.scala._
 import org.apache.flink.table.planner.runtime.FileSystemITCaseBase
 import org.apache.flink.table.planner.runtime.utils.{AbstractExactlyOnceSink, StreamingTestBase, TestingAppendSink, TestSinkUtil}
+import org.apache.flink.testutils.junit.extensions.parameterized.NoOpTestExtension
 import org.apache.flink.types.Row
 
-import org.junit.{Assert, Before, Test}
-import org.junit.Assert.assertEquals
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.{BeforeEach, Test}
+import org.junit.jupiter.api.extension.ExtendWith
 
 import scala.collection.{mutable, Seq}
 
 /** Streaming [[FileSystemITCaseBase]]. */
+@ExtendWith(Array(classOf[NoOpTestExtension]))
 abstract class StreamFileSystemITCaseBase extends StreamingTestBase with FileSystemITCaseBase {
```

Comment:
Emmm, @Before
  @BeforeEach
  def before(): Unit = {
    this.env = StreamExecutionEnvironment.getExecutionEnvironment
    env.setParallelism(4)
    if (enableObjectReuse) {
      this.env.getConfig.enableObjectReuse()
    }
    val setting = EnvironmentSettings.newInstance().inStreamingMode().build()
    this.tEnv = StreamTableEnvironment.create(env, setting)
  }

Reply:
I did notice this issue, but I found that the `lifecycle` method like `before` in `StreamingTestBase ` double annotationed with `@Before` and `@BeforeEach`.
```
  @Before
  @BeforeEach
  def before(): Unit = {
    this.env = StreamExecutionEnvironment.getExecutionEnvironment
    env.setParallelism(4)
    if (enableObjectReuse) {
      this.env.getConfig.enableObjectReuse()
    }
    val setting = EnvironmentSettings.newInstance().inStreamingMode().build()
    this.tEnv = StreamTableEnvironment.create(env, setting)
  }
```

Reply:
And these two `Rules` are not used in this PR related subclass
```
@Rule
def thrown: ExpectedException = expectedException
@Rule
def tempFolder: TemporaryFolder = _ tempFolder
```

Reply:
Of course, in the long run, we should also migrate it to `Junit5`. Perhaps we can create a new ticket to track it because it is also a significant workload.

Reply:
agree about rules, `@after/@before` annotations.
Another point is that `StreamingTestBase` extends another junit4 based `AbstractTestBase` which has rule 
```java
@ClassRule
    public static final MiniClusterWithClientResource MINI_CLUSTER_RESOURCE =
            new MiniClusterWithClientResource(
                    new MiniClusterResourceConfiguration.Builder()
                            .setNumberTaskManagers(1)
                            .setNumberSlotsPerTaskManager(DEFAULT_PARALLELISM)
                            .build());
```
it looks like this rule is not required for streaming tests (`before`/`after` is enough) 
please correct me if I'm wrong

Reply:
To some extent, yes. Because we manually set the parallelism in `before` method:
```
env.setParallelism(4)
```

Reply:
thanks for clarification